src/components/forms/FormGenerator.tsx

```diff
@@ -0,0 +1,266 @@
+import React, { useEffect, useState } from "react"
+import { Formik, Form } from "formik"
+import {
+  IFormVariable,
+  IFormData,
+  IFormValidationData,
+} from "../../utils/types"
+import { groupVariables, formValidationSchema } from "../../utils/forms"
+import FieldsCreator from "./FieldsCreator"
+import { FORM_LABEL_DISPLAY_OPTIONS } from "../../utils/data"
+
+interface IFormGeneratorProps {
+  formVariables: IFormVariable[]
+  initialFormData: IFormData
+  //formValidationData: IFormValidationData
+  handleProgress: Function
+  handleCurrentStep: Function
+}
+
+type ITargetName = string
+type ITargetValue = string | number | boolean
+type ITarget = {
+  target: { name: ITargetName; value: ITargetValue }
+}
+
+const FormGenerator: React.FC<IFormGeneratorProps> = ({
+  formVariables,
+  initialFormData,
+  //formValidationData,
+  handleProgress,
+  handleCurrentStep,
+}) => {
+  // const navigate = useNavigate()
+  // const { t } = useTranslation()
+  const [step, setStep] = useState(0)
+  //const [completed, setCompleted] = useState(false)
+  const [changedTarget, setChangedTarget] = useState<ITarget | null>(null)
+  const [showVars, setShowVars] = useState(0)
+  const [currentVarsDataAppend, setCurrentVarsDataAppend] = useState<
+    IFormVariable[]
+  >([])
+  const [formValues, setFormValues] = useState<IFormData>()
+  const [countAppendDependent, setCountAppendDependent] = useState(0)
+
+  const handleChange = (e: ITarget) => {
+    setChangedTarget(e)
+  }
+
+  const groupedVariableList = groupVariables(formVariables)
+  const currentVarsData = groupedVariableList[step + 1]
+
+  const formValidationData: IFormValidationData = formValidationSchema(
+    currentVarsDataAppend,
+  )
+
+  // Question append implementation
+  const handleValueChange = (values: IFormData) => {
+    setFormValues(values)
+  }
+  const allNonDependsElement = currentVarsData.filter(
+    (currentVarData) => !currentVarData.choosenOption,
+  )
+  const allDependsElement = currentVarsData.filter(
+    (currentVarData) => currentVarData.choosenOption,
+  )
+
+  const handleTargetValueChange = (
+    targetName: ITargetName,
+    targetValue: ITargetValue,
+  ) => {
+    formatAppendQuestion(targetName, targetValue)
+  }
+
+  const handleChangeInput = (e: ITarget) => {
+    formatAppendQuestion(e.target.name, e.target.value)
+  }
+
+  const formatAppendQuestion = (
+    targetName: ITargetName,
+    targetValue: ITargetValue,
+  ) => {
+    if (targetValue) {
```

Comment:
Instead of nesting deeply, invert this check

Example
```ts
if (!targetValue) {
  return null
}
```

But you'll also want to be careful for legitimate values of `0`, because that simple truthy/falsy check above will fail.